litestar/utils/sync.py

```diff
@@ -78,8 +107,39 @@ async def _async_generator(self) -> AsyncGenerator[T, None]:
             except ValueError:
                 return
 
-    def __aiter__(self) -> AsyncIteratorWrapper[T]:
+    def __aiter__(self) -> AsyncGeneratorWrapper[T, S]:
         return self
 
     async def __anext__(self) -> T:
         return await self.generator.__anext__()
+
+    async def aclose(self) -> None:
+        if isinstance(self._original_generator, Generator):
+            while self._is_in_thread:
+                pass
```

Comment:
Please remove this code and the other instances like it. I don't think we should implicitly try to make (async) generators thread safe. Instead, we should just handle them correctly